### PR TITLE
add: Await InBlock and  Broadcast Status When Sending Extrinsic

### PR DESCRIPTION
**Pull request by developerfred**
#65 

### update jenkins
https://github.com/scs/substrate-api-client/pull/74

### CODE REVIEW

**Comment by brenzi** [comment]
@developerfred thanks for your PR. May I ask you to fix the unit test?

**Comment by developerfred** [comment]
@brenzi I will fix

**Comment by brenzi** [comment]
It could be that the error actually comes from testing against the wrong node in jenkins. The api-client should now actually test against our node-template #71

**Comment by developerfred** [comment]
@brenzi I'll take a look at the issue #71

**Comment by brenzi** [comment]
@developerfred I'v manually tested your PR. It builds, but it doesn't work. Here's how to reproduce:

```
https://github.com/scs/substrate-node-template.git
git checkout test-api-client
cargo build --release
./target/release/node-template purge-chain --dev
./target/release/node-template --dev

# other terminal
https://github.com/developerfred/substrate-api-client.git
git checkout await-InBlock 
cargo build --release --examples
RUST_LOG=debug ./target/release/examples/example_transfer
```
The log should show you that the client waits for `finalized`

now, change the example:
```
diff --git a/src/examples/example_transfer.rs b/src/examples/example_transfer.rs
index 094cbf6..4ae73e5 100755
--- a/src/examples/example_transfer.rs
+++ b/src/examples/example_transfer.rs
@@ -47,7 +47,7 @@ fn main() {
 
     // send and watch extrinsic until finalized
     let tx_hash = api
-        .send_extrinsic(xt.hex_encode(), XtStatus::Finalized)
+        .send_extrinsic(xt.hex_encode(), XtStatus::InBlock)
         .unwrap();
     println!("[+] Transaction got finalized. Hash: {:?}\n", tx_hash);
```
I would expect it to wait for `InBlock` now, but it only awaits `Ready`.
That's because you're calling the exact same function that we wrote to wait for `ready`. I'll comment in your commit

File: src/lib.rs [hunk 1]
```diff
@@ -344,6 +344,18 @@ where
                 info!("ready: {}", res);
                 Ok(None)
             }
+            XtStatus::Broadcast => {
+                rpc::send_extrinsic(self.url.clone(), jsonreq, result_in);
```
Comment (brenzi) [inline src/lib.rs:348]: you can't call this function because that is hard-coded to wait for `ready`.

File: src/lib.rs [hunk 2]
```diff
@@ -344,6 +344,18 @@ where
...
+                let res = result_out.recv().unwrap();
+                info!("broadcast: {}", res);
+                Ok(None)
+            }
+            XtStatus::InBlock => {
+                rpc::send_extrinsic(self.url.clone(), jsonreq, result_in);
```
Comment (brenzi) [inline src/lib.rs:354]: same here

File: src/lib.rs (hunk 1 above)
Comment (developerfred) [reply src/lib.rs:348]: I will add a new await function

File: src/lib.rs (hunk 2 above)
Comment (developerfred) [reply src/lib.rs:354]: I will add a new await function

**Comment by developerfred** [comment]
@brenzi fixed pull on https://github.com/scs/substrate-api-client/commit/934414657436f507df53ba07b6e33978968d7798

File: src/rpc/client.rs [hunk 3]
```diff
@@ -150,7 +189,16 @@ fn parse_status(msg: &str) -> (XtStatus, Option<String>) {
...
                     (XtStatus::Finalized, Some(hash.to_string()))
-                } else {
+                }
+                else if let Some(hash) = obj.get("inBlock") {
+                    info!("inBlock: {:?}", hash);
+                    (XtStatus::InBlock, Some(hash.to_string()))
+                }
+                else if let Some(hash) = obj.get("broadcast") {
+                    info!("Broadcast: {:?}", hash);
+                    (XtStatus::Broadcast, Some(hash.to_string()))                     (XtStatus::InBlock, Some(hash.to_string()))
```
[the remark this answers is not in the document]
Comment (developerfred) [reply src/rpc/client.rs:199]: sorry @brenzi !

**Comment by brenzi** [comment]
I'm sorry that I have to disappoint you @developerfred but we do prefer to wait for a task until it is well done and reviewing it when it is tested and ready

**Pull request closed without merging**